### PR TITLE
Add workflow for publishing to pypi and creating a release

### PR DESCRIPTION
**Pull request by dennismeister93**
This PR adds a workflow for publishing to pypi and creating a release

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.10"
+    - name: Install pypa/build
+      run: >-
+        python3 -m
+        pip install
+        build
+        --user
```
Comment (MP91) [inline .github/workflows/release.yml:44]:
```suggestion
      run:  pip install build --user
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    - name: Store the distribution packages
+      uses: actions/upload-artifact@v4
+      with:
+        name: python-package-distributions
+        path: dist/
+
+  publish-to-pypi:
+    name: >-
+      Publish Python 🐍 distribution 📦 to PyPI
+    if: startsWith(github.ref, 'refs/tags/')
```
Comment (MP91) [inline .github/workflows/release.yml:56]: not needed, due to the triggers

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    - name: Sign the dists with Sigstore
+      uses: sigstore/gh-action-sigstore-python@v2.1.1
+      with:
+        inputs: >-
+          ./dist/*.tar.gz
+          ./dist/*.whl
+    - name: Create release
+      id: create_release
+      uses: softprops/action-gh-release@v2
+      if: startsWith(github.ref, 'refs/tags/')
```
Comment (MP91) [inline .github/workflows/release.yml:101]: not needed, due to the triggers

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+name: Publish Python 🐍 distribution 📦 to PyPI
+
+on:
+  workflow_dispatch: # input version manually. Overrides push tag
+    inputs:
+      tag:
+        description: "Release version"
+        required: true
+        default: "v0.0.0"
```
Comment (MP91) [inline .github/workflows/release.yml:23]: when should we trigger this manually? would vote for removal, this should just run when releasing a new version

**Review by MP91 — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 4 above)
Comment (doosuu) [reply .github/workflows/release.yml:23]: Second this. Manual release should not be possible (for now) - I don't see the use case

Comment (dennismeister93) [reply .github/workflows/release.yml:23]: fine for me

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      name: pypi
+      url: https://pypi.org/p/velocitas-lib
+      id-token: write
+
+    steps:
+    - name: Download dists folder
+      uses: actions/download-artifact@v4
+      with:
+        name: python-package-distributions
+        path: dist/
```
[the remark this answers is not in the document]
Comment (dennismeister93) [reply .github/workflows/release.yml:70]: doesnt hurt and looks better structured

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        packages-dir: python-package/dist
+
+  github-release:
+    name: >-
+      Sign the Python 🐍 distribution 📦 with Sigstore
+      and upload them to GitHub Release
+    needs:
+    - publish-to-pypi
```
Comment (MP91) [inline .github/workflows/release.yml:74]: not really needed, pypi and github release can run in parallel

**Review by MP91 — APPROVED** [review]
lgtm

**Pull request merged**